### PR TITLE
Add GetLanguageID function

### PR DESCRIPTION
**Pull request by look**
The Linguist-defined language IDs are important to our use case because they are used as database identifiers. This adds a new generator to extract the language IDs into a map and uses that to implement `GetLanguageID`.

Because one language has the ID 0, there is no way to tell if a language name is found or not. If desired, we could add this by returning `(string, bool)` from `GetLanguageID`. But none of the other functions that take language names do this,
so I didn't want to introduce it here.

We've been using this code in production using an internal fork for a while, and I would love to get it upstreamed. ❤️

### CODE REVIEW

**Review by JnBrymn — APPROVED** [review]
This seems good to me!

**Review by elireisman — APPROVED** [review]
This looks great thanks for updating it 👍

File: common.go [hunk 1]
```diff
@@ -490,6 +490,14 @@ func GetLanguageExtensions(language string) []string {
 	return data.ExtensionsByLanguage[language]
 }
 
+// GetLanguageID returns the ID for the language. IDs are assigned by GitHub.
+//
+// NOTE: The zero value (0) is a valid language ID, so this API has no way to
```
Comment (lafriks) [inline common.go:495]: Maybe it would be better to return -1 for invalid languages?

File: common.go [hunk 2]
```diff
@@ -490,6 +490,14 @@ func GetLanguageExtensions(language string) []string {
...
+// distinguish an invalid language name and a match on the language with ID 0.
+func GetLanguageID(language string) int {
+	return data.IDByLanguage[language]
```
Comment (lafriks) [inline common.go:498]: Something like this:
```suggestion
	id, ok := data.IDByLanguage[language]
	if !ok {
		return -1
	}
	return id
```

File: common.go (hunk 1 above)
Comment (look) [reply common.go:495]: I would be open to this. I was concerned about Enry not being in control of these IDs, but talked to @lildude about it and he pointed out that the [IDs are now generated by hashing the name](https://github.com/github/linguist/blob/e6860ab64b463e022294b0f71d468d5243687027/script/update-ids#L8-L10), so a negative number is not possible.

However, it doesn't feel very Go-like. I would probably prefer to do it the way that [map access works](https://golang.org/doc/effective_go#maps) and return `(int, bool)`.

**Review by bzz — COMMENTED** [review]
Overall, looks great, thank you @look!

> The Linguist-defined language IDs are important to our use case

Would be interesting to know a bit more about your use case.

File: common.go (hunk 1 above)
Comment (bzz) [reply common.go:495]: > I would probably prefer to do it the way that map access works and return (int, bool).

Sounds very reasonable. Will you submit the patch as well?

**Review by bzz — APPROVED** [review]

File: common.go (hunk 1 above)
Comment (lafriks) [reply common.go:495]: (int, bool) also is ok with me :)

Comment (look) [reply common.go:495]: Thanks for the feedback 🙇 I'll submit an update to this PR that adjusts the return type.

**Review by look — COMMENTED** [review]
Our use case is that we index the language ID in a search index and use the value for filtering results by language.

**Review by lafriks — APPROVED** [review]

**Review by bzz — APPROVED** [review]
Looks great to me

**Pull request merged**